### ISSUE TITLE
Upgrade ownCloud 8.2.1 to Nextcloud 9.0.54 Issue

### ISSUE THREAD

**Issue by kneiser** [nextcloud/server#1669]
### Steps to reproduce
1. Delete the old owncloud log file in the data directory.
2. Backup owncloud data, config, and apps directories.
3. Rename the /var/www/html/owncloud directory to "xowncloud".
4. Download the Nextcloud zip file to user "Downloads" directory and extract.
5. Rename the extracted nextcloud directory to owncloud. Nextcloud is to be used as a "drop in" replacement for ownCloud.
6. Copy extracted owncloud directory to the /var/www/html directory.
7. Restore the ownCloud config.php file.
8. Restore the ownCloud database from backups.
9. Set recommended "strict" permissions and ownership on /var/www/html/owncloud directory and its' subdirectories.
10. Trigger the update process from the Nextcloud web interface.
11. The Nextcloud database and Official apps appear to be updated as shown in the web interface.
12. An error occurs and informs user to report the problem to the Nextcloud Community.

### Expected behaviour
A clean upgrade from ownCloud 8.2.1 to Nextcloud 9.0.x with no critical errors on the web interface. BTW, this same problem exists when upgrading from ownCloud 8.2.1 to 9.0.2. In the recent past I tried to do updates in the ownCloud 8.2.x > 8.2.1 and kept getting errors, such as webDav not correctly running. According to documentation sources I should be able to upgrade from 8.2.1 to Nextcloud 9.

I raised the following issue on the same subject, someone felt it was not worth reviewing further and closed it: https://github.com/nextcloud/server/issues/297

See also issue raised by another person: https://github.com/owncloud/core/issues/23867

Poking around the forum and ownCloud issues this is **NOT THE FIRST TIME** this was raised. Does anyone have any idea what the problem is? **PLEASE?**

### Actual behavior
The following error is displayed in the Nextcloud web interface:

```
federation: An exception occurred while executing 'CREATE TABLE `oc_trusted_servers` (`id` INT AUTO_INCREMENT NOT NULL, `url` VARCHAR(512) NOT NULL COMMENT 'Url of trusted server', `url_hash` VARCHAR(255) DEFAULT '' NOT NULL COMMENT 'sha1 hash of the url without the protocol', `token` VARCHAR(128) DEFAULT NULL COMMENT 'token used to exchange the shared secret', `shared_secret` VARCHAR(256) DEFAULT NULL COMMENT 'shared secret used to authenticate', `status` INT DEFAULT 2 NOT NULL COMMENT 'current status of the connection', `sync_token` VARCHAR(512) DEFAULT NULL COMMENT 'cardDav sync token', UNIQUE INDEX url_hash (`url_hash`), PRIMARY KEY(`id`)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB': SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'oc_trusted_servers' already exists
```

PLEASE NOTE:

Upon reading past forum items it was suggested by some users to drop the table in the error and repeat the upgrade process. This was done then another error of the same was reported but a different table name is displayed. If you keep repeating this eventually tables with production data are named. This is where I stopped upgrading. I am not willing to lose valuable data from the previous ownCloud production instance.

Is the upgrade process trying to create the table a second time? Or is it as a result of some sort of table consistency test that makes sure all required tables for the upgrade exist? Is there an SQL script the upgrade process uses where this error is originating?

<details>
<summary>Click here for: Tables in ownCloud (before upgrade)</summary>

```
oc_activity
oc_activity_mq
oc_addressbookchanges
oc_addressbooks
oc_appconfig
oc_book_bookmark
oc_book_location
oc_book_scales
oc_bookmarks
oc_bookmarks_tags
oc_calendarchanges
oc_calendarobjects
oc_calendars
oc_calendarsubscriptions
oc_cards
oc_cards_properties
oc_clndr_calendars
oc_clndr_objects
oc_clndr_repeat
oc_clndr_share_calendar
oc_clndr_share_event
oc_comments
oc_comments_read_markers
oc_contacts_addressbooks
oc_contacts_cards
oc_contacts_cards_properties
oc_credentials
oc_dav_shares
oc_documents_invite
oc_documents_member
oc_documents_op
oc_documents_revisions
oc_documents_session
oc_external_applicable
oc_external_config
oc_external_mounts
oc_external_options
oc_file_locks
oc_file_map
oc_filecache
oc_files_trash
oc_group_admin
oc_group_user
oc_groups
oc_jobs
oc_lucene_status
oc_mimetypes
oc_mounts
oc_music_albums
oc_music_ampache_sessions
oc_music_ampache_users
oc_music_artists
oc_music_playlist_tracks
oc_music_playlists
oc_music_tracks
oc_notifications
oc_ownnote
oc_ownnote_parts
oc_preferences
oc_privatedata
oc_properties
oc_quicknotes_colors
oc_quicknotes_notes
oc_quicknotes_tasks
oc_schedulingobjects
oc_share
oc_share_external
oc_storages
oc_systemtag
oc_systemtag_object_mapping
oc_trusted_servers
oc_users
oc_vcategory
oc_vcategory_to_object

```
</details>

<details>
<summary>Click here for: Tables in Nextcloud (after upgrade attempt)</summary>

```
oc_activity	
oc_activity_mq	
oc_addressbookchanges	
oc_addressbooks	
oc_admin_sections	
oc_admin_settings	
oc_appconfig	
oc_book_bookmark	
oc_book_location	
oc_book_scales	
oc_bookmarks	
oc_bookmarks_tags	
oc_bruteforce_attempts	
oc_calendarchanges	
oc_calendarobjects	
oc_calendars	
oc_calendarsubscriptions	
oc_cards	
oc_cards_properties	
oc_clndr_calendars	
oc_clndr_objects	
oc_clndr_repeat	
oc_clndr_share_calendar	
oc_clndr_share_event	
oc_comments	
oc_comments_read_markers	
oc_contacts_addressbooks	
oc_contacts_cards	
oc_contacts_cards_properties	
oc_credentials	
oc_dav_shares	
oc_documents_invite	
oc_documents_member	
oc_documents_op	
oc_documents_revisions	
oc_documents_session	
oc_external_applicable	
oc_external_config	
oc_external_mounts	
oc_external_options	
oc_federated_reshares	
oc_file_locks	
oc_filecache	
oc_files_trash	
oc_flow_checks	
oc_flow_operations	
oc_group_admin	
oc_group_user	
oc_groups	
oc_jobs	
oc_lucene_status	
oc_mimetypes	
oc_mounts	
oc_music_albums	
oc_music_ampache_sessions	
oc_music_ampache_users	
oc_music_artists	
oc_music_playlist_tracks	
oc_music_playlists	
oc_music_tracks	
oc_notifications	
oc_ownnote	
oc_ownnote_parts	
oc_preferences	
oc_privatedata	
oc_properties	
oc_quicknotes_colors	
oc_quicknotes_notes	
oc_quicknotes_tasks	
oc_schedulingobjects	
oc_share	
oc_share_external	
oc_storages	
oc_systemtag	
oc_systemtag_group	
oc_systemtag_object_mapping	
oc_trusted_servers
oc_users	
oc_vcategory	
oc_vcategory_to_object	

```
</details>

<details>
<summary>Click here for: Repeating Upgrade Errors</summary>
```
An exception occurred while executing 'CREATE TABLE `oc_notifications` (`notification_id` INT AUTO_INCREMENT NOT NULL, `app` VARCHAR(32) NOT NULL, `user` VARCHAR(64) NOT NULL, `timestamp` INT DEFAULT 0 NOT NULL, `object_type` VARCHAR(64) NOT NULL, `object_id` VARCHAR(64) NOT NULL, `subject` VARCHAR(64) NOT NULL, `subject_parameters` LONGTEXT DEFAULT NULL, `message` VARCHAR(64) DEFAULT NULL, `message_parameters` LONGTEXT DEFAULT NULL, `link` VARCHAR(4000) DEFAULT NULL, `actions` LONGTEXT DEFAULT NULL, INDEX oc_notifications_app (`app`), INDEX oc_notifications_user (`user`), INDEX oc_notifications_timestamp (`timestamp`), INDEX oc_notifications_object (`object_type`, `object_id`), PRIMARY KEY(`notification_id`)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB': SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'oc_notifications' already exists

An exception occurred while executing 'CREATE TABLE `oc_addressbooks` (`id` BIGINT UNSIGNED AUTO_INCREMENT NOT NULL, `principaluri` VARCHAR(255) DEFAULT NULL, `displayname` VARCHAR(255) DEFAULT NULL, `uri` VARCHAR(255) DEFAULT NULL, `description` VARCHAR(255) DEFAULT NULL, `synctoken` INT UNSIGNED DEFAULT 1 NOT NULL, UNIQUE INDEX addressbook_index (`principaluri`, `uri`), PRIMARY KEY(`id`)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB': SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'oc_addressbooks' already exists

AT THIS POINT I STOPPED AS THE TABLE ABOVE ALREADY HAS PRODUCTION DATA.

```
</details>
### Server configuration
**Operating system**

Fedora 23 - 32-bit

**Web server:**

Apache 2.4.18

**Database:**

mariadb 10.0.23

**PHP version:**

5.6.22

**Nextcloud version:** (see Nextcloud admin page)
9.0.54

**Updated from an older Nextcloud/ownCloud or fresh install:**

ownCloud 8.2.1

**Where did you install Nextcloud from:**

As indicated in "Steps to reproduce".

**Signing status:**
<details>
<summary>Signing status</summary>

```
Could not find these results.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - admin_audit: 1.0.0
  - comments: 0.2
  - dav: true
  - external: 1.2
  - federatedfilesharing: 0.1.0
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_videoplayer: 0.9.8
  - gallery: 14.5.0
  - password_policy: 1.0.0
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - theming: 0.2.0
  - updatenotification: 0.1.0
Disabled:
  - activity
  - bookmarks
  - calendar
  - contacts
  - delete_confirmation
  - encryption
  - federation
  - files_mv
  - files_versions
  - firstrunwizard
  - logreader
  - music
  - notifications
  - ownnote
  - quicknotes
  - user_external
  - user_ldap
  - user_saml
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "oc5zrx7pkm6x",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "192.168.0.244"
        ],
        "datadirectory": "\/var\/files\/owncloud\/data",
        "overwrite.cli.url": "http:\/\/192.168.0.244\/owncloud",
        "dbtype": "mysql",
        "version": "8.2.1.4",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "updatechecker": true,
        "has_internet_connection": true,
        "loglevel": 0,
        "log_rotate_size": 26214400,
        "log_rotate_sizerotation": 209715200,
        "trashbin_retention_obligation": "14, auto",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "theme": "",
        "maintenance": false,
        "appstore.experimental.enabled": true
    }
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

local

**Are you using encryption:** yes/no

no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

none

### Client configuration
**Browser:**

Firefox 49.0

**Operating system:**

Fedora 24 - 32-bit

#### Web server error log
<details>
<summary>Web server error log</summary>
```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
```
{"reqId":"V-mL1Mvtpk8QGANmBVkMtAAAAAo","remoteAddr":"192.168.0.163","app":"core","message":"starting upgrade from 8.2.1.4 to 9.0.54.5","level":0,"time":"2016-10-09T00:14:12+00:00","method":"GET","url":"\/owncloud\/core\/ajax\/update.php?requesttoken=PQA5LQRLFmcEG0Ufd3sWMQJfUmIQAH8LIC1wIwkZIm0%3D%3Ay5uWn3C0GQ%2B%2FFOPfA11Vr%2F6orEAVKTE7DroZtAW2sos%3D","user":"--"}
{"reqId":"V-mL1Mvtpk8QGANmBVkMtAAAAAo","remoteAddr":"192.168.0.163","app":"PHP","message":"file_put_contents(\/var\/files\/owncloud\/data\/.htaccess): failed to open stream: Permission denied at \/var\/www\/html\/owncloud\/lib\/private\/setup.php#486","level":3,"time":"2016-10-09T00:14:12+00:00","method":"GET","url":"\/owncloud\/core\/ajax\/update.php?requesttoken=PQA5LQRLFmcEG0Ufd3sWMQJfUmIQAH8LIC1wIwkZIm0%3D%3Ay5uWn3C0GQ%2B%2FFOPfA11Vr%2F6orEAVKTE7DroZtAW2sos%3D","user":"--"}
{"reqId":"V-mL1Mvtpk8QGANmBVkMtAAAAAo","remoteAddr":"192.168.0.163","app":"federation","message":"Exception: {\"Exception\":\"Doctrine\\\\DBAL\\\\Exception\\\\TableExistsException\",\"Message\":\"An exception occurred while executing 'CREATE TABLE `oc_trusted_servers` (`id` INT AUTO_INCREMENT NOT NULL, `url` VARCHAR(512) NOT NULL COMMENT 'Url of trusted server', `url_hash` VARCHAR(255) DEFAULT '' NOT NULL COMMENT 'sha1 hash of the url without the protocol', `token` VARCHAR(128) DEFAULT NULL COMMENT 'token used to exchange the shared secret', `shared_secret` VARCHAR(256) DEFAULT NULL COMMENT 'shared secret used to authenticate', `status` INT DEFAULT 2 NOT NULL COMMENT 'current status of the connection', `sync_token` VARCHAR(512) DEFAULT NULL COMMENT 'cardDav sync token', UNIQUE INDEX url_hash (`url_hash`), PRIMARY KEY(`id`)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB':\\n\\nSQLSTATE[42S01]: Base table or view already exists: 1050 Table 'oc_trusted_servers' already exists\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(116): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(945): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'CREATE TABLE `o...')\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/mdb2schemamanager.php(168): Doctrine\\\\DBAL\\\\Connection->query('CREATE TABLE `o...')\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/mdb2schemamanager.php(71): OC\\\\DB\\\\MDB2SchemaManager->executeSchemaChange(Object(Doctrine\\\\DBAL\\\\Schema\\\\Schema))\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db.php(173): OC\\\\DB\\\\MDB2SchemaManager->createDbFromStructure('\\\/var\\\/www\\\/html\\\/o...')\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/installer.php(588): OC_DB::createDbFromStructure('\\\/var\\\/www\\\/html\\\/o...')\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/installer.php(558): OC_Installer::installShippedApp('federation')\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/updater.php(336): OC_Installer::installShippedApps(true)\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/updater.php(214): OC\\\\Updater->doUpgrade('9.0.54.5', '8.2.1.4')\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/core\\\/ajax\\\/update.php(121): OC\\\\Updater->upgrade()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":49}","level":3,"time":"2016-10-09T00:15:18+00:00","method":"GET","url":"\/owncloud\/core\/ajax\/update.php?requesttoken=PQA5LQRLFmcEG0Ufd3sWMQJfUmIQAH8LIC1wIwkZIm0%3D%3Ay5uWn3C0GQ%2B%2FFOPfA11Vr%2F6orEAVKTE7DroZtAW2sos%3D","user":"--"}
{"reqId":"V-mL1Mvtpk8QGANmBVkMtAAAAAo","remoteAddr":"192.168.0.163","app":"notifications","message":"Exception: {\"Exception\":\"Doctrine\\\\DBAL\\\\Exception\\\\TableExistsException\",\"Message\":\"An exception occurred while executing 'CREATE TABLE `oc_notifications` (`notification_id` INT AUTO_INCREMENT NOT NULL, `app` VARCHAR(32) NOT NULL, `user` VARCHAR(64) NOT NULL, `timestamp` INT DEFAULT 0 NOT NULL, `object_type` VARCHAR(64) NOT NULL, `object_id` VARCHAR(64) NOT NULL, `subject` VARCHAR(64) NOT NULL, `subject_parameters` LONGTEXT DEFAULT NULL, `message` VARCHAR(64) DEFAULT NULL, `message_parameters` LONGTEXT DEFAULT NULL, `link` VARCHAR(4000) DEFAULT NULL, `actions` LONGTEXT DEFAULT NULL, INDEX oc_notifications_app (`app`), INDEX oc_notifications_user (`user`), INDEX oc_notifications_timestamp (`timestamp`), INDEX oc_notifications_object (`object_type`, `object_id`), PRIMARY KEY(`notification_id`)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB':\\n\\nSQLSTATE[42S01]: Base table or view already exists: 1050 Table 'oc_notifications' already exists\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(116): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(945): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'CREATE TABLE `o...')\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/mdb2schemamanager.php(168): Doctrine\\\\DBAL\\\\Connection->query('CREATE TABLE `o...')\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/mdb2schemamanager.php(71): OC\\\\DB\\\\MDB2SchemaManager->executeSchemaChange(Object(Doctrine\\\\DBAL\\\\Schema\\\\Schema))\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db.php(173): OC\\\\DB\\\\MDB2SchemaManager->createDbFromStructure('\\\/var\\\/www\\\/html\\\/o...')\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/installer.php(588): OC_DB::createDbFromStructure('\\\/var\\\/www\\\/html\\\/o...')\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/installer.php(558): OC_Installer::installShippedApp('notifications')\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/updater.php(336): OC_Installer::installShippedApps(true)\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/updater.php(214): OC\\\\Updater->doUpgrade('9.0.54.5', '8.2.1.4')\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/core\\\/ajax\\\/update.php(121): OC\\\\Updater->upgrade()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":49}","level":3,"time":"2016-10-09T00:15:18+00:00","method":"GET","url":"\/owncloud\/core\/ajax\/update.php?requesttoken=PQA5LQRLFmcEG0Ufd3sWMQJfUmIQAH8LIC1wIwkZIm0%3D%3Ay5uWn3C0GQ%2B%2FFOPfA11Vr%2F6orEAVKTE7DroZtAW2sos%3D","user":"--"}
{"reqId":"V-mL1Mvtpk8QGANmBVkMtAAAAAo","remoteAddr":"192.168.0.163","app":"dav","message":"Exception: {\"Exception\":\"Doctrine\\\\DBAL\\\\Exception\\\\TableExistsException\",\"Message\":\"An exception occurred while executing 'CREATE TABLE `oc_addressbooks` (`id` BIGINT UNSIGNED AUTO_INCREMENT NOT NULL, `principaluri` VARCHAR(255) DEFAULT NULL, `displayname` VARCHAR(255) DEFAULT NULL, `uri` VARCHAR(255) DEFAULT NULL, `description` VARCHAR(255) DEFAULT NULL, `synctoken` INT UNSIGNED DEFAULT 1 NOT NULL, UNIQUE INDEX addressbook_index (`principaluri`, `uri`), PRIMARY KEY(`id`)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB':\\n\\nSQLSTATE[42S01]: Base table or view already exists: 1050 Table 'oc_addressbooks' already exists\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(116): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(945): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'CREATE TABLE `o...')\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/mdb2schemamanager.php(168): Doctrine\\\\DBAL\\\\Connection->query('CREATE TABLE `o...')\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/mdb2schemamanager.php(71): OC\\\\DB\\\\MDB2SchemaManager->executeSchemaChange(Object(Doctrine\\\\DBAL\\\\Schema\\\\Schema))\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db.php(173): OC\\\\DB\\\\MDB2SchemaManager->createDbFromStructure('\\\/var\\\/www\\\/html\\\/o...')\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/installer.php(588): OC_DB::createDbFromStructure('\\\/var\\\/www\\\/html\\\/o...')\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/installer.php(558): OC_Installer::installShippedApp('dav')\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/updater.php(336): OC_Installer::installShippedApps(true)\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/updater.php(214): OC\\\\Updater->doUpgrade('9.0.54.5', '8.2.1.4')\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/core\\\/ajax\\\/update.php(121): OC\\\\Updater->upgrade()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":49}","level":3,"time":"2016-10-09T00:15:18+00:00","method":"GET","url":"\/owncloud\/core\/ajax\/update.php?requesttoken=PQA5LQRLFmcEG0Ufd3sWMQJfUmIQAH8LIC1wIwkZIm0%3D%3Ay5uWn3C0GQ%2B%2FFOPfA11Vr%2F6orEAVKTE7DroZtAW2sos%3D","user":"--"}
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by nickvergessen** [comment]
Duplicate of https://github.com/nextcloud/server/issues/761

Do you mind adding your information there?

**Comment by kneiser** [comment]
@nickvergessen I looked at issue 761 and noticed you added a link to this posted issue. Is that enough?

**Comment by kneiser** [comment]
@nickvergessen @jospoortvliet  Back Recently this issue (#1669) was closed and referenced in issue #761. Today I noticed that #761 has been closed by the original issue author. I reviewed the closure item and another related issue. I don't argee that the solution for #761 would resolve my issue concerns. I have in the past posted this same issue (#297) and it was closed with no solution. My reported issues have existed since ownCloud 9 which was inherited in to the Nextcloud fork. As you will see in my issues the problem stems from existing tables that exist in an owncloud 8.2.x instance that already contains tables that will also be in the new Nextcloud instance target. For some reason the Nextcloud upgrade process is trying to recreate these existing tables and thus flagging them as critical errors thus failing the upgrade process. So why is the process doing this? I have never modified the database instances and have faithfully followed the update/upgrade steps has referenced in your documentation. I have been using this product since owncloud 4.x and cannot upgrade without having to completely redo my installation from scratch.

Issue #1669 is a reposting of issue #297, the same issue. I am VERY sad (to the point of frustration) that my issues keep getting closed with out any solution, meaningful work around, or reasonable response to the problem. Other community users have had the same issues and from what I have researched and observed (issue and forum items) they have had to resort to starting from sratch. People with large data collections are NOT able to upgrade. It is tiresome of having to repost the same issue over and over again.

Having this same issue come up with every upgrade is a major road block from an upgrade. I like Nextcloud and want to continue to use it.
